src/ProxyClient/Cloudflare.php

```diff
@@ -52,7 +52,7 @@ class Cloudflare extends HttpProxyClient implements ClearCapable, PurgeCapable,
     public function __construct(
         Dispatcher $httpDispatcher,
         array $options = [],
-        RequestFactory $messageFactory = null
+        RequestFactoryInterface $messageFactory = null
```

Comment:
to avoid BC breaks, we should remove the type declaration and instead have a phpdoc that does `RequestFactory|RequestFactoryInterface`